packages/openid4vc/src/openid4vc-issuer/OpenId4VcIssuerService.ts

```diff
@@ -356,49 +403,80 @@ export class OpenId4VcIssuerService {
       'urn:ietf:params:oauth:grant-type:pre-authorized_code': {
         'pre-authorized_code':
           preAuthorizedCodeFlowConfig.preAuthorizedCode ?? (await agentContext.wallet.generateNonce()),
+        // v11 only
         user_pin_required: preAuthorizedCodeFlowConfig.userPinRequired ?? false,
+        tx_code: preAuthorizedCodeFlowConfig.txCode,
       },
     }
 
     return grants
   }
 
   private findOfferedCredentialsMatchingRequest(
+    agentContext: AgentContext,
     credentialOffer: OpenId4VciCredentialOfferPayload,
     credentialRequest: OpenId4VciCredentialRequest,
-    credentialsSupported: OpenId4VciCredentialSupported[],
+    credentialsSupported: OpenId4VciCredentialSupportedWithId[],
     issuanceSession: OpenId4VcIssuanceSessionRecord
-  ): OpenId4VciCredentialSupportedWithId[] {
-    const offeredCredentials = getOfferedCredentials(credentialOffer.credentials, credentialsSupported)
-
-    return offeredCredentials.filter((offeredCredential) => {
-      if (offeredCredential.format !== credentialRequest.format) return false
-      if (issuanceSession.issuedCredentials.includes(offeredCredential.id)) return false
-
-      if (
-        credentialRequest.format === OpenId4VciCredentialFormatProfile.JwtVcJson &&
-        offeredCredential.format === credentialRequest.format
-      ) {
-        return equalsIgnoreOrder(offeredCredential.types, credentialRequest.types)
-      } else if (
-        credentialRequest.format === OpenId4VciCredentialFormatProfile.JwtVcJsonLd &&
-        offeredCredential.format === credentialRequest.format
-      ) {
-        return equalsIgnoreOrder(offeredCredential.types, credentialRequest.credential_definition.types)
-      } else if (
-        credentialRequest.format === OpenId4VciCredentialFormatProfile.LdpVc &&
-        offeredCredential.format === credentialRequest.format
-      ) {
-        return equalsIgnoreOrder(offeredCredential.types, credentialRequest.credential_definition.types)
-      } else if (
-        credentialRequest.format === OpenId4VciCredentialFormatProfile.SdJwtVc &&
-        offeredCredential.format === credentialRequest.format
-      ) {
-        return offeredCredential.vct === credentialRequest.vct
+  ): OpenId4VciCredentialConfigurationsSupported {
+    const offeredCredentialsData = isCredentialOfferV1Draft13(credentialOffer)
+      ? credentialOffer.credential_configuration_ids
+      : credentialOffer.credentials
+
+    const proofTypesSupported = getProofTypesSupported(agentContext)
+    const offeredCredentials = getOfferedCredentials(offeredCredentialsData, credentialsSupported, {
+      proofTypesSupported,
+    })
```

Comment:
Same here, using all alg_values supported as proof_types supported gives inconsistent results with what the isuser actually wants